watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1175 @@
+            raise RuntimeError(
+                "Invalid flash_method: {flash_method}. Use one of {', '.join(valid_flashes)}"
+            )
+
+        clone = deepcopy(json_input)
+        for k, v in survey.items():
+            d = clone["params"]
             if flash_method == "wateranalysis":
-                for param in modified_clone["params"]["waterAnalysisInputs"]:
-                    if param["name"] in survey[i].keys():
-                        param.update({"value": survey[i][param["name"]]})
-            elif flash_method == "isothermal":
-                for param in survey[i]:
-                    if param in ["Temperature", "Pressure"]:
-                        modified_clone["params"][param.lower()]["value"] = survey[i][
-                            param
-                        ]
-                    elif param in modified_clone["params"]["inflows"]["values"]:
+                d = d["waterAnalysisInputs"]
+                for param in d:
+                    if param["name"].lower() == k.lower():
                         if self.relative_inflows:
-                            modified_clone["params"]["inflows"]["values"][
-                                param
-                            ] += survey[i][param]
+                            param["value"] += v[index]
                         else:
-                            modified_clone["params"]["inflows"]["values"][
-                                param
-                            ] = survey[i][param]
-                    else:
-                        raise ValueError(
-                            "Only composition and temperature/pressure surveys are currently supported."
+                            param["value"] = v[index]
+                        _logger.info(
+                            f"Updating {k} for sample #{index} clone: new value = {param['value']}"
                         )
             else:
-                raise ValueError(
-                    " Only 'wateranalysis' and 'isothermal' currently supported."
+                if k in d:
+                    pass
+                elif k in d["inflows"]["values"]:
+                    d = d["inflows"]["values"]
+                elif k in d["corrosionParameters"]:
+                    d = d["corrosionParameters"]
+                else:
+                    _logger.warning(f"Survey key {k} not found in JSON input.")
+                if self.relative_inflows:
+                    if isinstance(d[k], dict):
+                        d[k]["value"] += v[index]
+                        val = d[k]["value"]
+                    else:
+                        d[k] += v[index]
+                        val = d[k]
+                else:
+                    if isinstance(d[k], dict):
+                        d[k]["value"] += v[index]
+                        val = d[k]["value"]
+                    else:
+                        d[k]["value"] += v[index]
+                        val = d[k]["value"]
```

Comment:
Two things: (1) `if not self.relative_inflows`, shouldn't the value be equal instead of additive? And (2) it looks like you accidentally kept in reference to a "value" key for types that are not dicts. Here's the suggestion to reflect these items:
```suggestion
                    if isinstance(d[k], dict):
                        d[k]["value"] = v[index]
                        val = d[k]["value"]
                    else:
                        d[k] = v[index]
                        val = d[k]
```

Reply:
Yes, thanks for catching those!